src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -112,11 +112,12 @@ object FilterIndexRule
         val allIndexes = indexManager.getIndexes(Seq(Constants.States.ACTIVE))
 
         val candidateIndexes = allIndexes.filter { index =>
-          indexCoversPlan(
-            outputColumns,
-            filterColumns,
-            index.indexedColumns,
-            index.includedColumns)
+          index.derivedDataset.isInstanceOf[CoveringIndex] &&
```

Comment:
Pattern matching can be used here. It will resolved later down the cast with `asInstanceOf[...]`.

Reply:
Done